server/lobbyconnection.py

```diff
@@ -965,32 +978,35 @@ async def command_game_join(self, message):
             game = self.game_service[uuid]
         except KeyError:
             await self.send({
-                "command": "notice",
+                "command": "game_join_failed",
                 "style": "info",
-                "text": "The host has left the game."
+                "text": "The host has left the game.",
+                "uid": uuid
             })
             return
 
         if self.player.id in game.host.foes:
-            raise ClientError("You cannot join games hosted by this player.")
+            raise GameJoinError("You cannot join games hosted by this player.", uuid)
 
         if not game or game.state is not GameState.LOBBY:
             self._logger.debug("Game not in lobby state: %s state %s", game, game.state)
             await self.send({
-                "command": "notice",
+                "command": "game_join_failed",
                 "style": "info",
-                "text": "The game you are trying to join is not ready."
+                "text": "The game you are trying to join is not ready.",
+                "uid": uuid
             })
             return
 
         if game.init_mode != InitMode.NORMAL_LOBBY:
-            raise ClientError("The game cannot be joined in this way.")
+            raise GameJoinError("The game cannot be joined in this way.", uuid)
```

Comment:
Similar thing here, I think `ClientError` is correct since this would mean that the client is trying to join a matchmaker game as if it were a custom game, which means there is a bug in the client. The client should never have to handle this error message because it should never be acting in such a way that it gets triggered in the first place.